parcels/particleset.py

```diff
@@ -71,23 +71,16 @@ class ParticleSet(object):
 
     :param grid: Grid object from which to sample velocity
     :param pclass: Optional class object that defines custom particle
-    :param lon: Optional list of initial longitude values for particles
-    :param lat: Optional list of initial latitude values for particles
-    :param start: Optional starting point for initilisation of particles
-                 on a straight line. Use start/finish instead of lat/lon.
-    :param finish: Optional end point for initilisation of particles on a
-                 straight line. Use start/finish instead of lat/lon.
-    :param start_field: Optional field for initialising particles stochastically
-                 according to the presented density field. Use instead of lat/lon.
-    :param size: Optional initial size of particle set, only required when using
-                 start/finish or start_field arguments
+    :param lon: List of initial longitude values for particles
+    :param lat: List of initial latitude values for particles
     """
 
-    def __init__(self, grid, pclass=JITParticle, lon=None, lat=None,
-                 start=None, finish=None, start_field=None, size=None):
+    def __init__(self, grid, pclass=JITParticle, lon=None, lat=None):
+        # Convert numpy arrays to one-dimensional lists
+        lon = lon.flatten() if isinstance(lon, np.ndarray) else lon
+        lat = lat.flatten() if isinstance(lat, np.ndarray) else lat
+        size = len(lon)
         self.grid = grid
```

Comment:
Given PARCELS current lack of smart grid loading, at present when executing the same `ParticleSet` object across multiple `grid` objects, you must either update this pointer yourself (which I think works, but manually changing pointers always makes me nervous!) or create new `ParticleSet` objects and copy particle properties one-by-one between calls to `execute()`.
Should we think about adding a `ParticleSet.switch_grid()` function that intelligently switches this pointer to a new grid object, whilst checking that all particles are still in the domain, keeping the original time_origin, and updating any appropriate `Particle` variables which may rely on grid (such as `xi` and `yi` for `JITParticle` objects) etc?